lib/gremlex/client.ex

```diff
@@ -53,10 +53,15 @@ defmodule Gremlex.Client do
 
       error ->
         Logger.error("Error establishing connection to server: #{inspect(error)}")
-        GenServer.start_link(__MODULE__, %{}, [])
+        GenServer.start_link(__MODULE__, nil, [])
```

Comment:
Consider returning `{:stop, "Socket is missing"}` right here to be more explicit. `init()` won't be needed then.

Reply:
[`init/1`](https://hexdocs.pm/elixir/GenServer.html#c:init/1) is required callback for GenServer

Reply:
OK, it is indeed. Good to go then.